File: z-clients/javascript/src/models/accessRule.ts
// this file is @generated
import {
    type AccessRuleEffect,
    AccessRuleEffectSerializer,
} from './accessRuleEffect';

export interface AccessRule {
    effect: AccessRuleEffect;
    resource: string;
    actions: string[];
}

export const AccessRuleSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AccessRule {
        return {
            effect: AccessRuleEffectSerializer._fromJsonObject(object['effect']),
            resource: object['resource'],
            actions: object['actions'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AccessRule): any {
        return {
            'effect': AccessRuleEffectSerializer._toJsonObject(self.effect),
            'resource': self.resource,
            'actions': self.actions,
        };
    }
}

File: z-clients/rust/src/models/access_rule.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{access_rule_effect::AccessRuleEffect, resource_pattern::ResourcePattern};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AccessRule {
    pub effect: AccessRuleEffect,

    pub resource: ResourcePattern,

    pub actions: Vec<String>,
}

impl AccessRule {
    pub fn new(effect: AccessRuleEffect, resource: ResourcePattern, actions: Vec<String>) -> Self {
        Self {
            effect,
            resource,
            actions,
        }
    }
}
